016/master

BenGitsCode opened the pull request:
This PR addresses #27  #28 #24 #21 

It may only close #24 and #21 as the other two are a little more subjective.

BenGitsCode committed:
Update from talk-template

-  Refresh for cohort 016

BenGitsCode committed:
Clarify instructions in 'Implementing a feature'

- Close #22
- Specify `origin development` in git pull rebase.

BenGitsCode committed:
Clarify that branches are feature-based

-  Close #23
-  Add clarification into general guidelines section.

BenGitsCode committed:
Update Deployment steps for browser-template

-  changes to bundle and vendor in gitignore are no longer
necessary.

BenGitsCode committed:
Change instances of group to team

-  Some of these are so semantic it seems silly, but the official
name of the "group" project will be changed to "team" to
specifically reflect the project goal (teamwork).

-  See ga-wdi-boston/group-project/issues/229

BenGitsCode committed:
Add d3-based git app to additional resources

-  Very simple d3 app that visualizes the actions of things like
merge, rebase, diff, checkout, etcetera.
-  I've never seen someone use my advice of "afraid of git?" just
clone your repo, remove the remote if you're so afraid of breaking
things and then play with git to see what your commands do. I'm
hoping that providing this very low-stakes sandbox, people will
actually experiment with different git commands.
-  May tie into a small demo in the talk itself.

BenGitsCode committed:
Update initial git branch diagram

-  Also altered the language to reflect the diagram
  -  Dev and feature instead of feature and master.

BenGitsCode committed:
Update new commit on dev diagram

-  Altered language to reflect the new diagram.
  - Dev and feature instead of feature and master.

BenGitsCode committed:
Update merge development into feature diagram

-  Update diagram with new image link.
-  Adjust language accordingly.

BenGitsCode committed:
Update rebase development onto feature

-  Update diagram.
-  Adjust language accordingly
  -  Master => Development.
- #27

BenGitsCode committed:
Remove Rebase final diagram

-  Previous diagram accomplishes the same goal of showing the
original commits being destroyed as new SHAs are generated for
the new commits on rebased branch.

BenGitsCode committed:
Rephrased rebase danger warning and SHA change

-  The italicized "remember" section could be worded better, but
I think it's an important point to add.
-  The developers won't grok rebasing (or any git action) without
a minimally viable mental model of what they actually do.
-  Plan to further clarify this with a short "demystification"
addition.

BenGitsCode committed:
Minor rephrasing of instructional note

-  Simple rephrasing for clarity.

BenGitsCode committed:
Add brief new lab to README

-  Lab asks students to use d3 app to perform merges and rebases
with visual demonstration of operations.
-  Intention is to give developers interaction with the commands
and seeing what they actually do, before breaking down the rebase
operation with a diagram.

BenGitsCode committed:
Correct final lab Header

-  `Project Workflow` were on new line, they should be part
of the header.

BenGitsCode committed:
Add rebase sequence of operations diagram

-  Diagram showing the sequence of operations performed in a
rebase.
-  Making git diagrams is hard. This isn't perfect but it's a
start.

BenGitsCode committed:
Fix typo and grammar of Lab headers

-  Overview was misspelled.
-  Labs should be Lab: foo not Lab : foo.

BenGitsCode committed:
Update JS-Template instructions

-  Use browser-template.
-  Close #24

BenGitsCode committed:
Enhance instructors note

-  Also added language from #21 into appropriate places along
the rebase in pictures section.
-  Close #21

raq929 commented on README.md:
```diff
@@ -57,10 +57,10 @@ Today, we'll introduce a third: _rebasing_.
 ### Git Rebase, in Pictures
 
-Suppose that you have two branches in your project, `master` and `feature`,
+Suppose that (in addition to master) you have two branches in your project, `dev` and `feature`,
```
I would prefer development, but OK

BenGitsCode replied:
I agree, I only went with dev because development obscured some of the diagram with its length. I'll open an issue for it now.

raq929 commented on README.md:
```diff
@@ -60,12 +60,12 @@ Today, we'll introduce a third: _rebasing_.
 
-If you were to check out the master branch and make a new commit,
- the `feature` branch would no longer point to the end of the `master` branch.
+If you were to check out the dev branch and make a new commit,
```
Needs bactics around `dev`

raq929 commented on README.md:
```diff
@@ -96,6 +96,18 @@ However, as long as you're only rebasing your own code on top of things,
  it's a great way of making sure that `feature` stays up to date. _Remember: when you "rebase your code on top of things" the branch following `git rebase` is what you're rebasing your branch "on top of" — it will be the new "base" for your current branch if executed._
 
+### Lab : Identify the differences between rebase and merge.
+- Open [Explain Git with D3](https://onlywei.github.io/explain-git-with-d3/) in your browser.
```
❤️

raq929 commented on README.md:
```diff
@@ -247,8 +247,8 @@ To practice the workflow we've prescribed for you,
-You may start by downloading the JS-Template as a ZIP
- and dropping those files into your repo.
+You may start by downloading the [Browser-Template](https://github.com/ga-wdi-boston/browser-template) as a ZIP
+ and renaming and moving those files into your repo.
```
I think this needs to be more specific. ie, YOU WILL LOSE HIDDEN FILES UNLESS YOU DO IT X WAY.

raq929 commented on README.md:
```diff
@@ -90,7 +90,7 @@ Git actually destroys the old commits and replaces them with new commits
 This is one of the things that can make `git rebase` dangerous,
  and it's the reason why you never rebase code that's already been
- published and shared - you run the risk of breaking other peoples' code.
+ published and shared - you run the risk of breaking other peoples' code. After a rebase, pushing to the branch will require --force-with-lease since the history has been rewritten.
```
Add Antony's language " rebase adds commits onto a shared commit in both branches history and then reapplies your commits on top of those." 
I don't think they should be taught how to force push, I think it will lead to plroblems.

raq929 replied:
Actually, this is included below. Remove the 'After a rebase code here.' I certainly don't want to reinforce the -force stuff twice. So remove it here, keep it below.

BenGitsCode replied:
Removed in c05ed8b

BenGitsCode committed:
Remove repetitive --force-with-lease note

- this is included below. Remove the 'After a rebase code here.' I
certainly don't want to reinforce the -force stuff twice.
- per @raq929 comments on ga-wdi-boston/git-workflow-teams#29

BenGitsCode committed:
Address PR comments about clarifying language used

- per @raq929 comments on ga-wdi-boston/git-workflow-teams#29
